community/contributing/pr_review_guidelines.rst

```diff
@@ -0,0 +1,357 @@
+.. _doc_pr_review_guidelines:
+
+Pull request review process
+===========================
+
+.. note::
+
+    This page is intended to provide insight into the pull request (PR) review
+    process that we aspire to. As such, it is primarily targeted at engine
+    maintainers who are responsible for reviewing and approving pull requests.
+    That being said, much of the content is useful for prospective contributors
+    wanting to know how to ensure that their PR is merged.
+
+From a high level, the ideal life cycle of a pull request looks like: 
+  1. A contributor opens a PR that fixes a specific problem.
+
+  2. Other contributors provide feedback on the PR (including reviewing and/or
+     approving the PR, as appropriate).
+
+  3. An engine maintainer reviews the code and provides feedback, requests
+     changes, or approves the pull request, as appropriate.
+
+  4. Another maintainer reviews the code with a focus on code style/clarity and
+     approves it once satisfied.
+
+  5. A team leader or a member of the production team merges the pull request if
+     satisfied that it has been sufficiently reviewed.
+
+This document will explain steps 2, 3, 4, and 5 in more detail. For a more
+detailed explanation of the pull request workflow please see the :ref:`pull
+request workflow document <doc_pr_workflow>`. 
+
+.. note:: 
+  In practice these steps may blend together. Oftentimes maintainers will
+  provide comments on code style and code quality at the same time and will
+  approve a pull request for both.
+
+Typically the first interaction on a pull request will be an engine maintainer
+assigning tags to the pull request and flagging it for review by someone
+familiar with that area of code.
+
+Engine maintainers are folks who are "members" of the Godot project repository
+on Github and/or are listed on the `Teams page <https://godotengine.org/teams>_`
+on the Godot website. Maintainers are typically responsible for a given area of
+the engine. Typically this means they are the people who are given more trust to
+approve and recommend pull requests for merging.
```

Comment:
Nit: Has 'typically' twice in 2 sentences. Maybe replace one with 'often' or 'most of the time' or similar.